exporters/trace/src/main/java/com/google/cloud/opentelemetry/trace/TraceExporter.java

```diff
@@ -15,134 +15,84 @@
-        new CloudTraceClientImpl(TraceServiceClient.create(stub)),
-        configuration.getAttributeMapping(),
-        configuration.getFixedAttributes());
-  }
-
-  private static TraceExporter createWithClient(
-      String projectId,
-      CloudTraceClient cloudTraceClient,
-      ImmutableMap<String, String> attributeMappings,
-      Map<String, AttributeValue> fixedAttributes) {
-    return new TraceExporter(projectId, cloudTraceClient, attributeMappings, fixedAttributes);
+  /**
+   * Method that generates an instance of {@link TraceExporter} using a minimally configured {@link
+   * TraceConfiguration} object that requires no input from the user. Since no project ID is
+   * specified, default project ID is used instead. See {@link ServiceOptions#getDefaultProjectId()}
+   * for details.
+   *
+   * <p>This method defers the initialization of an actual {@link TraceExporter} to a point when it
+   * is actually needed - which is when the spans need to be exported. As a result, while this
+   * method does not throw any exception, an exception may still be thrown during the attempt to
+   * generate the actual {@link TraceExporter}.
+   *
+   * @return A configured instance of {@link TraceExporter} which gets initialized lazily once
+   *     {@link TraceExporter#export(Collection)} is called.
+   */
+  public static SpanExporter createWithDefaultConfiguration() {
+    return new TraceExporter(TraceConfiguration.builder().build());
   }
 
-  TraceExporter(
-      String projectId,
-      CloudTraceClient cloudTraceClient,
-      ImmutableMap<String, String> attributeMappings,
-      Map<String, AttributeValue> fixedAttributes) {
-    this.projectId = projectId;
-    this.cloudTraceClient = cloudTraceClient;
-    this.projectName = ProjectName.of(projectId);
-    this.translator = new TraceTranslator(attributeMappings, fixedAttributes);
+  /**
+   * Method that generates an instance of {@link TraceExporter} using a {@link TraceConfiguration}
+   * that allows the user to provide custom configuration for Traces.
+   *
+   * <p>This method defers the initialization of an actual {@link TraceExporter} to a point when it
+   * is actually needed - which is when the spans need to be exported. As a result, while this
+   * method does not throw any exception, an exception may still be thrown during the attempt to
+   * generate the actual {@link TraceExporter}.
+   *
+   * @param configuration The {@link TraceConfiguration} object that determines the user preferences
+   *     for Trace.
+   * @return An instance of {@link TraceExporter} as a {@link SpanExporter} object
+   */
+  public static SpanExporter createWithConfiguration(TraceConfiguration configuration) {
+    return new TraceExporter(configuration);
   }
 
   @Override
   public CompletableResultCode flush() {
-    // We do no exporter buffering of spans, so we're always flushed.
-    return CompletableResultCode.ofSuccess();
+    return internalTraceExporterSupplier.get().flush();
```

Comment:
Having a `NoopSpanExporter` facilitates for a much cleaner and simple looking code, a downside for this is that if for some reason `flush()` or `shutdown()` is called prior to a call to `export()`, the current code would first initialize the `internalTraceExporter` - only to flush or shutdown. 

We could use an `AtomicBoolean` to keep track if its initialized and conditionally skip initializing;  simply returning success codes, but this would make code a less elegant here.